### PR TITLE
Feature Selection

### PR DESCRIPTION
**Pull request by bertubezz**

### CODE REVIEW

File: terms/featureselection_ab.tex [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+\begin{figure}
+  \includegraphics{feature_selection/growth.png}
+  \caption{Growth of the number of features in the UCI ML repository. Reproduced from \citet{sammut2017encyclopedia}.}
+  \label{fig:fs_growth}
+\end{figure}
+
+\citet{guyon2003introduction} states that the objective of FS is three-fold: improving the prediction performance of the predictors, providing faster and more cost-effective predictors, and providing a better understanding of the underlying process that generated the data. Furthermore, the aim the FS is to select features that are \textit{independent} of each other but \textit{highly dependent} on the value to be predicted.
```
Comment (jp-um) [inline terms/featureselection_ab.tex]: "state" not "states" there are two authors

File: terms/featureselection_ab.tex [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+% For example, considering feature $f1$ as shown in Figure \ref{fig:feature_selection}a, it can be seen that $f1$ is a relevant feature because it can distinguish between the classes. Feature $f2$ (Figure \ref{fig:feature_selection}b) on the other hand is unable to distinguish between the two classes and is therefore irrelevant for the classification and hence discarded.  As seen in Figure \ref{fig:feature_selection}c, feature $f6$ is also a relevant feature because it perfectly distinguishes between the two classes.  However, $f6$ does not add any new information that $f1$ has not already provided, so it is said to be a redundant feature. Feature $f4$ (Figure \ref{fig:feature_selection}d) is partly able to distinguish between the two classes.  The overlapping instances are either caused by errors in the data (noise) or due to the nature of the attribute.  This type of feature is said to be noisy and is also discarded.
```
Comment (jp-um) [inline terms/featureselection_ab.tex]: Feature id should be a subscript $f_{1}$, for f1, f2 etc.

File: terms/featureselection_ab.tex [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+%   \caption{Example plots showing (a) a relevant feature, (b) an irrelevant feature, (c) a redundant feature and (d) a noisy feature. Reproduced from \citet{sammut2017encyclopedia}.}
+%   \label{fig:feature_selection}
+% \end{figure}
+
+
+FS keeps the original feature, that is, it maintains the physical meanings of the original feature without any transformation. Furthermore, \citet{masaeli2010transformation} found that this property has its significance in many practical applications such as finding relevant genes to a specific disease and building a sentiment lexicon for sentiment analysis.
+
+According to \citet{zhao2010advancing}, FS is an optimization problem that can be split into two perspective; (1) searching for the best subset of features, (2) principle model for selecting, adding removing or changing new features during the search using a certain evaluation measure. Mainly there are three models used for FS: filter, wrapper and embedded.
+
+\section{Filter Models}\label{sec:fs_filter}\index{feature selection!filter models}
```
Comment (jp-um) [inline terms/featureselection_ab.tex]: "Feature Selection" in index should be capitalized accordingly

File: terms/featureselection_ab.tex [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  \item \textit{dependency measure}\index{feature selection!dependency measure}: is also known as correlation measure or similarity measure. It measures the ability to predict the value of one variable from the value of another. In FS for classification, we look for how strongly a feature is associated with the class. A feature $f1$ is preferred to another feature $f2$ if the association between feature $f1$ and class $C$ is higher than the association between $f2$ and $C$. In FS for clustering, the association between two random features measures the similarity between the two \citep{liu2005toward}. Two common dependency measure functions are \textit{Pearson's correlation} and \text{Bhattacharyya dependence}\citep{de2015feature}.
```
Comment (jp-um) [inline terms/featureselection_ab.tex]: Distance capital D -- consistency.

File: terms/featureselection_ab.tex [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+\section{Embedded models}\label{sec:fs_filter}\index{feature selection!embedded models}
+Some learning algorithms have in-built mechanisms to deal with feature selection \citep{guyon2003introduction,sammut2017encyclopedia}. This makes the selection of the features more efficient than the wrapper method and is specifically designed for the algorithm of choice. Another improvement on the other methods is that whilst the filter and wrapper methods select the features as part of the pre-processing of the data, the embedded models cater for feature selection as part of the learning process. One such example is the use of \textit{l1-norm} regularisation in \textit{Support Vector Machines} (SVM) \citep{guyon2003introduction,de2015feature}. Here the coefficients of less important features are reduced to zeros thus eliminating them. \textit{L2-norm} regularisation can also be used for feature selection but was found to be less effective than \textit{l1-norm} regularisation \citep{de2015feature}. \textit{Decision trees} also have a feature selection strategy embedded within the algorithm \citep{garcia2015data}.
```
Comment (jp-um) [inline terms/featureselection_ab.tex]: Good.

**Review by jp-um — CHANGES_REQUESTED** [review]
- Not bad but you need equations to explain some concepts rather than plough through everything
- Not a single note on automated feature selection??  how it is done.  Important.

**Comment by bertubezz** [comment]
By automated feature selection are you referring to streaming feature selection?

Thanks.

**Comment by jp-um** [comment]
Automated feature selection is just that.  Using automated ways how to select the best features to build an ML model.  Example using a Genetic Algorithm to find the best performat set of features.

**Comment by bertubezz** [comment]
I have updated my section with your suggestion and resolved the issues that you mentioned.

Thanks

**Comment by bertubezz** [comment]
Furthermore, now I am using \nameref to reference sections instead of '_below_'.

**Comment by jp-um** [comment]
Merged, thanks.

**Pull request merged**